src/patches/mapkey.cc

```diff
@@ -0,0 +1,143 @@
+#include "config.h"
+#include "prime/EventSystem.h"
+#include "utils.h"
+#include <absl/strings/ascii.h>
+#include <absl/strings/str_split.h>
+#include <cstdint>
+#include <iostream>
+#include <key.h>
+#include <map>
+#include <mapkey.h>
+#include <modifierkey.h>
+#include <prime/TMP_InputField.h>
+#include <string>
+#include <string_view>
+
+MapKey::MapKey()
+{
+  this->Key = KeyCode::None;
+}
+
+MapKey* MapKey::Parse(std::string_view key)
+{
+  auto strippedKey = absl::StripTrailingAsciiWhitespace(key);
+  auto lowerKey    = absl::AsciiStrToUpper(strippedKey);
+  auto wantedKeys  = absl::StrSplit(lowerKey, "-", absl::SkipWhitespace());
+
+  auto mapKey = new MapKey();
```

Comment:
I would personally prefer this to just return a std::optional instead of a raw pointer that is never deleted. If that's not desired, at least a unique_ptr :)

Reply:
Not sure I follow this, but that may be down to my limited experience with this style of c++